pkg/network/server.go

```diff
@@ -277,6 +277,17 @@ func (s *Server) handleBlockCmd(p Peer, block *core.Block) error {
 	return s.bQueue.putBlock(block)
 }
 
+// handlePing processes ping request.
+func (s *Server) handlePing(p Peer, ping *payload.Ping) error {
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, payload.NewPing(s.id, s.chain.HeaderHeight())))
+}
+
+// handlePing processes pong request.
+func (s *Server) handlePong(p Peer, pong *payload.Ping) error {
+	// do nothing on pong.
```

Comment:
There has to be some outstanding pings management and we should check to see whether this pong is expected or not. We also need some method to send pings probably, though I'm not sure what would be the best trigger for them.

Reply:
Maybe if the node doesn't send anything in some amount of time like N number of blocks then it's time to ping it and disconnect if there is no reply in M seconds.

Reply:
Yeah that could be the case. BTW though the neo doc and neo # node didn't find that there is any way to invoke `ping` there =/

Reply:
I think this mechanism is used more in NEO 3.0, there were some discussions around that, though not sure.